### PR TITLE
fix!: change the default delimiter of `$str$concat` from `"-"` to `""` and `ignore_nulls` should be a named argument

### PR DESCRIPTION
**Pull request by eitsupi**
Ref pola-rs/polars#13690

### CODE REVIEW

File: R/expr__string.R [hunk 1]
```diff
@@ -215,25 +215,33 @@ ExprStr_len_chars = function() {
...
+#' Vertically concatenate the string values in the column to a single string value.
 #'
-#' @description Vertically concatenate the values in the Series to a single
-#' string value.
 #' @param delimiter The delimiter to insert between consecutive string values.
-#' @param ignore_nulls Ignore null values. If `FALSE`, null values will be
+#' @param ... Ignored.
+#' @param ignore_nulls Ignore null values (default). If `FALSE`, null values will be
 #' propagated: if the column contains any null values, the output is null.
 #' @keywords ExprStr
```
Comment (etiennebacher) [inline R/expr__string.R:224]:
```suggestion
```

File: R/expr__string.R [hunk 2]
```diff
@@ -215,25 +215,33 @@ ExprStr_len_chars = function() {
...
 #' # Series list of strings to Series of concatenated strings
-#' df = pl$DataFrame(list(bar = list(c("a", "b", "c"), c("1", "2", NA))))
-#' df$select(pl$col("bar")$list$eval(pl$col("")$str$concat("-"))$list$first())
-ExprStr_concat = function(delimiter = "-", ignore_nulls = TRUE) {
+#' df = pl$DataFrame(bar = list(c("a", "b", "c"), c("d", "f", NA)))
+#' df
+#'
+#' df$select(
+#'   pl$col("bar")$list$eval(pl$element()$str$concat("-"))$list$first()
+#' )
```
Comment (etiennebacher) [inline R/expr__string.R:240]: this should probably be an example for `list$eval()` instead?

**Review by etiennebacher — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 3]
```diff
@@ -15,6 +15,10 @@
         that are either column names or regular expressions (#852).
     -   There is no argument for `pl$len()`. If you want to measure the length of
         specific columns, you should use `pl$count(...)` (#852).
+    -   `<Expr>$str$concat()` method's `delimiter` argument's default value is
+        changed from `"-"` to `""` (#853).
```
[the remark this answers is not in the document]
Comment (eitsupi) [reply NEWS.md:19]: I would like to change now as it is too much of a hassle to split the update into two separate updates.

**Review by etiennebacher — APPROVED** [review]
Thanks!

**Pull request merged**